crates/ide_assists/src/handlers/extract_struct_from_enum_variant.rs

```diff
@@ -159,10 +178,95 @@ fn existing_definition(db: &RootDatabase, variant_name: &ast::Name, variant: &Va
         .any(|(name, _)| name.to_string() == variant_name.to_string())
 }
 
+struct ExtractedGenerics {
+    lifetimes: Vec<ast::LifetimeParam>,
+    types: Vec<ast::TypeParam>,
+    consts: Vec<ast::ConstParam>,
```

Comment:
Do we need the split here? This duplicates a lot of code paths 3 times, while we could handle just `ast::GenericParam` and match where needed for a specific one of them. As in instead of using `lifetime_params()`, `type_params()` and `const_params()` just use `generic_params()`